src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java

```diff
@@ -375,6 +376,7 @@ public void uploadCoreDeviceCAs(String thingName, X509Certificate certificate)
             // by the next catch block
             throw e;
         } catch (Exception e) {
+            domainEvent.emit(new ServiceErrorEvent());
```

Comment:
this isn't a service error. This is most likely a network or permissions error.

We need to distinguish between these types of expected and unexpected errors. For example, if the exception is an SDKException, then it is not a service error.